src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -145,15 +147,24 @@ void updateSimulationStats() {
 
     /** Write final simulation statistical data to db. */
     void writeSimulationStats() {
-        for (Beacon beacon : beacons) {
-            beaconsObservedSum.putIfAbsent(String.valueOf(beacon.getId()), 0D);
-        }
-        Map<String, Double> beaconsObservedPercent = beaconsObservedSum.entrySet().stream()
-                .collect(toImmutableMap(e -> e.getKey(), e -> e.getValue() / (currentRound - 1)));
+        Map<String, List<Integer>> observedIntervals = beaconsObservedIntervals.entrySet().stream()
+                .collect(toImmutableMap(e -> String.valueOf(e.getKey().getId()),
+                e -> e.getValue().stream().filter(i -> i.observed()).map(ObservedInterval::duration).collect(toImmutableList())));
```

Comment:
Please split this to one more line and indent the `collect()` arguments one under the other, to make the code easier to read.